### ISSUE TITLE
Crash when trying to extract Britanny (via GeoJSON) from France

### ISSUE THREAD

**Issue by tfardet** [osmcode/osmium-tool#271]
## What version of osmium-tool are you using?

```
osmium version 1.15.0
libosmium version 2.19.0
Supported PBF compression types: none zlib lz4
```

## What operating system version are you using?

Manjaro Linux, GCC 13.2.1 (osmium was compiled locally)

## Tell us something about your system

Laptop, 8 CPUs, 16 GB of RAM

## What did you do exactly?

osmium extract -p bretagne.geojson france.osm.pbf -o france-bretagne.osm.pbf 

## What did you expect to happen?

Same as for the other regions: to get a proper extract.

## What did happen instead?

```
/usr/include/c++/13.2.1/bits/stl_vector.h:1125: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = std::vector<osmium::Segment>; _Alloc = std::allocator<std::vector<osmium::Segment> >; reference = std::vector<osmium::Segment>&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.
zsh: IOT instruction (core dumped)  osmium extract -p bretagne.geojson  -o france-bretagne.osm.pbf
```

## What did you do to try analyzing the problem?

The geojson is attached, I compared it with other regions that worked to see if there were differences but I could not find anything obvious except that the number of polygons is much greater.

[bretagne.tar.gz](https://github.com/osmcode/osmium-tool/files/12447466/bretagne.tar.gz)

**Comment by joto** [comment]
This should be fixed by #269 I just merged. Please try with master whether this works for you.

**Comment by tfardet** [comment]
Yep, all good, thanks!
